From 06d38795536376048446581217bd7a5727c009e7 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Thu, 16 Dec 2010 11:11:59 +0800
Subject: [PATCH 132/280] ENGR00137058 ipu lib: set pos before FG enable

After kernel SRM update fix, position could be set before
FG enable.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index 58757ac..d97164c 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -1371,17 +1371,17 @@ again:
 		dbg(DBG_INFO, "fb phyaddr1 0x%x\n", ipu_priv_handle->output.o_minfo[1].paddr);
 		dbg(DBG_INFO, "fb phyaddr2 0x%x\n", ipu_priv_handle->output.o_minfo[2].paddr);
 
-		blank = FB_BLANK_UNBLANK;
-		if ( ioctl(ipu_priv_handle->output.fd_fb, FBIOBLANK, blank) < 0) {
-			dbg(DBG_ERR, "UNBLANK FB failed!\n");
-		}
-
 		if (ipu_priv_handle->output.fb_chan == MEM_FG_SYNC) {
 			if ( ioctl(ipu_priv_handle->output.fd_fb, MXCFB_SET_OVERLAY_POS,
 						&(output->fb_disp.pos)) < 0)
 				dbg(DBG_ERR, "Set FB position failed!\n");
 		}
 
+		blank = FB_BLANK_UNBLANK;
+		if ( ioctl(ipu_priv_handle->output.fd_fb, FBIOBLANK, blank) < 0) {
+			dbg(DBG_ERR, "UNBLANK FB failed!\n");
+		}
+
 	}
 err:
 	return ret;
-- 
1.8.0

